File: tempFiles/browserJavaToKotlin - Copy-old/src/main/java/com/luciasoft/browserjavatokotlin/multibrowser/MyMessageBox.java
package com.luciasoft.browser.multibrowser;

import android.app.AlertDialog;
import android.content.Context;

class MyMessageBox extends AlertDialog
{
    enum ButtonsType { Ok, OkCancel, YesNo };

    private MyMessageBox(Context context)
    {
        super(context);
    }

    static void show(Context context, String title, String message, ButtonsType buttons, OnClickListener onOkYesClick, OnClickListener onCancelNoClick)
    {
        if (title == null) title = "";
        if (message == null) message = "";

        MyMessageBox mb = new MyMessageBox(context);

        if (buttons == null) buttons = ButtonsType.Ok;

        boolean twoButtons = buttons == ButtonsType.OkCancel || buttons == ButtonsType.YesNo;

        mb.setTitle(title);
        mb.setMessage(message);
        mb.setCancelable(twoButtons);

        String button1Text = buttons == ButtonsType.YesNo ? "Yes" : "OK";
        String button2Text = buttons == ButtonsType.YesNo ? "No" : "Cancel";

        mb.setButton(-1, button1Text, onOkYesClick);
        if (twoButtons) mb.setButton(-2, button2Text, onCancelNoClick);

        mb.show();
    }
}

File: browser/src/main/java/com/luciasoft/browser/MyMessageBox.kt
package com.luciasoft.browser

import android.app.AlertDialog
import android.content.Context
import android.content.DialogInterface

internal class MyMessageBox private constructor(context: Context) : AlertDialog(context)
{
    internal enum class ButtonsType
    {
        Ok, OkCancel, YesNo
    }

    companion object
    {
        @JvmStatic
        fun show(
            context: Context,
            title: String = "",
            message: String = "",
            buttons: ButtonsType = ButtonsType.Ok,
            onOkYesClick: DialogInterface.OnClickListener? = null,
            onCancelNoClick: DialogInterface.OnClickListener? = null
        )
        {
            val mb = MyMessageBox(context)
            val twoButtons = buttons == ButtonsType.OkCancel || buttons == ButtonsType.YesNo
            mb.setTitle(title)
            mb.setMessage(message)
            mb.setCancelable(twoButtons)
            val button1Text = if (buttons == ButtonsType.YesNo) "Yes" else "OK"
            val button2Text = if (buttons == ButtonsType.YesNo) "No" else "Cancel"
            mb.setButton(-1, button1Text, onOkYesClick)
            if (twoButtons) mb.setButton(-2, button2Text, onCancelNoClick)
            mb.show()
        }
    }
}